doc/decisions/binding_implementation.md

```diff
@@ -0,0 +1,41 @@
+# Binding Implementation
+
+## Problem
+
+In the previous error concept it was very useful to generate macros as we often added new errors. The code generation itself, however, is quite complicated (C++ code that prints the code; which is also not ideal for cross compilation, see also https://github.com/ElektraInitiative/libelektra/issues/2814).
+
+## Constraints
+
+## Assumptions
+
+- Error codes do not change very often
+
+## Considered Alternatives
+
+- Migrate to a more modern and easier way to generate code with our mustache system, and let this generate the (mapping) code for all compiled languages (C, C++, Java, Rust, Go).
+  All we get out of this is the removal of `std::cout << ...` code from C++ but not much more.
+
+- Migrate to CMake code that generates such macros/classes (see also https://github.com/ElektraInitiative/libelektra/issues/2814)
+  Approach will have problems with more complex languages though
+
+## Decision
+
+Write down the few macros manually, and also manually write down exceptions for the language bindings (and also the mappings from Elektra's internal errors to nice errors specific for the languages)
+
+Since error codes and crucial parts Elektra's core implementation will not change often this is the best approach with the least error rate.
```

Comment:
```suggestion
Since error codes and crucial parts Elektra's core implementation will not change often this is the best approach with minimal effort.
```